tools/mo/openvino/tools/mo/moc_frontend/extractor.py

```diff
@@ -19,46 +19,63 @@ def decode_name_with_port(input_model: InputModel, node_name: str, framework="")
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
-    found_nodes = []
-    found_node_names = []
-
-    node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        found_node_names.append('Tensor:' + node_name)
-        found_nodes.append(node)
-
-
-    def try_get_node(model, name, framework):
-        node = model.get_place_by_operation_name(name)
-        if node:
-            return node
-        if framework == "onnx":
-            tensor = model.get_place_by_tensor_name(name)
-            if tensor:
-                return tensor.get_producing_operation()
-        return None
-
-
-    regexp_post = r'(.+):(\d+)'
-    match_post = re.search(regexp_post, node_name)
-    if match_post:
-        node_post = try_get_node(input_model, match_post.group(1), framework)
-        if node_post:
-            node_post = node_post.get_output_port(output_port_index=int(match_post.group(2)))
-            if node_post:
-                found_node_names.append(match_post.group(1))
-                found_nodes.append(node_post)
-
-    regexp_pre = r'(\d+):(.+)'
-    match_pre = re.search(regexp_pre, node_name)
-    if match_pre:
-        node_pre = try_get_node(input_model, match_pre.group(2), framework)
-        if node_pre:
-            node_pre = node_pre.get_input_port(input_port_index=int(match_pre.group(1)))
-            if node_pre:
-                found_node_names.append(match_pre.group(2))
-                found_nodes.append(node_pre)
 
+    def try_get_nodes(input_model, node_name):
+        found_nodes = []
+        found_node_names = []
+        if node_name.count(":"):
```

Comment:
I believe this ':' is not needed - it is covered in regexp

Reply:
It is not needed in general, but I added it to avoid 2 regex calls if there is no ":" in the node name but sure can be removed